impala/_thrift_api.py

```diff
@@ -183,27 +199,47 @@ def setGetCustomHeadersFunc(self, func):
 
   def refreshCustomHeaders(self):
     if self.__get_custom_headers_func:
-      self.__custom_headers = self.__get_custom_headers_func(self.getAuthCookie())
-
-  def setAuthCookie(self):
-    if self.__auth_cookie_names:
-      c = get_first_matching_cookie(self.__auth_cookie_names, self.path, self.headers)
-      if c:
-        self.__auth_cookie = c
-        self.__auth_cookie_expiry = get_cookie_expiry(c)
-
-  def getAuthCookie(self):
-    if self.__auth_cookie and self.__auth_cookie_expiry and \
-        self.__auth_cookie_expiry <= datetime.datetime.now():
-      self.__auth_cookie = None
-    return self.__auth_cookie
-
-  def isAuthCookieSet(self):
-    return self.__auth_cookie is not None
-
-  def deleteAuthCookie(self):
-    self.__auth_cookie = None
-    self.__auth_cookie_expiry = None
+      cookie_headers, has_auth_cookie = self.getHttpCookieHeaders()
+      self.__custom_headers = \
+          self.__get_custom_headers_func(cookie_headers, has_auth_cookie)
+
+  def getHttpCookieHeaders(self):
+    if (self.__http_cookie_dict is None) or not self.__are_matching_cookies_found:
+      return None, False
+    cookie_headers = []
+    has_auth_cookie = False
+    for cn, c_tuple in self.__http_cookie_dict.items():
+      if c_tuple.cookie:
+        if c_tuple.expiry_time and c_tuple.expiry_time <= datetime.datetime.now():
+          self.__http_cookie_dict[cn] = Cookie(cookie=None, expiry_time=None)
+        else:
+          cookie_header = c_tuple.cookie.output(attrs=['value'], header='').strip()
+          cookie_headers.append(cookie_header)
+          if self.__auth_cookie_names and cn in self.__auth_cookie_names:
```

Comment:
nit: if not has_auth_cookie  and self.__auth_cookie_names and cn in self.__auth_cookie_names:

Reply:
Fixed